taskfiles/lint-cpp.yaml

```
@@ -0,0 +1,94 @@
+version: "3"
+
+tasks:
+  cpp-check:
+    desc: "Runs the C++ linters."
+    cmds:
+      - task: "cpp-format-check"
+      - task: "cpp-static-check"
+
+  cpp-fix:
+    desc: "Runs the C++ linters to fix all formatting issues and perform static code analysis. "
+    cmds:
+      - task: "cpp-format-fix"
+      - task: "cpp-static-fix"
+
+  cpp-format-check:
+    desc: "Runs the C++ linters that identify formatting issues."
+    sources: &cpp_format_src_files
+      - "{{.ROOT_DIR}}/**/*.cpp"
+      - "{{.ROOT_DIR}}/**/*.h"
+      - "{{.ROOT_DIR}}/**/*.hpp"
+      - "{{.ROOT_DIR}}/**/*.inc"
```

```suggestion
      - "{{.G_YSTDLIB_CPP_SRC_DIR}}/**/*.cpp"
      - "{{.G_YSTDLIB_CPP_SRC_DIR}}/**/*.h"
      - "{{.G_YSTDLIB_CPP_SRC_DIR}}/**/*.hpp"
```
(headers need to be self-contained so we don't allow `.inc` anymore)